### PR TITLE
chore: use rudder-go-kit

### PR DESCRIPTION
**Pull request by atzoum**
# Description

Switching to `rudder-go-kit` for logging, config, stats et al.

## Notion Ticket

[Link](https://www.notion.so/rudderstacks/8aac9087df644365acdf64e28e290153?v=6e06b0a5ade24f0aa5ffe05dc2972e84&p=78869499a67f452f887689e0378906e2&pm=s)

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/transformer/transformer_test.go [hunk 1]
```diff
@@ -339,6 +339,7 @@ func mockProxyHandler(timeout time.Duration, code int, response string) *mux.Rou
 		_, err := w.Write([]byte(response))
 		if err != nil {
 			http.Error(w, fmt.Sprintf("Provided response is faulty, please check it. Err: %v", err.Error()), http.StatusInternalServerError)
+			return
```
Comment (fracasula) [inline router/transformer/transformer_test.go:342]: Nice :+1:

**Review by fracasula — APPROVED** [review]
Most boring review in RS history :smile: 
Thanks @atzoum :+1:

File: gateway/integration_test.go [hunk 2]
```diff
@@ -21,11 +21,13 @@ import (
 	"github.com/stretchr/testify/require"
 	"golang.org/x/sync/errgroup"
 
+	kitHelper "github.com/rudderlabs/rudder-go-kit/testhelper"
```
Comment (lvrach) [inline gateway/integration_test.go:24]: [minor] more idiomatic for package name to be lower case only
```suggestion
	kithelper "github.com/rudderlabs/rudder-go-kit/testhelper"
```

File: router/eventorder_test.go [hunk 3]
```diff
@@ -18,17 +18,18 @@ import (
...
+	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource"
 	"github.com/rudderlabs/rudder-server/router/utils"
 	"github.com/rudderlabs/rudder-server/runner"
 	"github.com/rudderlabs/rudder-server/testhelper/health"
 
 	"golang.org/x/sync/errgroup"
 
 	"github.com/ory/dockertest/v3"
-	"github.com/rudderlabs/rudder-server/testhelper"
+	kitHelper "github.com/rudderlabs/rudder-go-kit/testhelper"
```
Comment (lvrach) [inline router/eventorder_test.go:30]:
```suggestion
	kithelper "github.com/rudderlabs/rudder-go-kit/testhelper"
```

File: router/network.go [hunk 4]
```diff
@@ -134,10 +134,10 @@ func (network *NetHandleT) SendPost(ctx context.Context, structData integrations
 
 		req, err := http.NewRequestWithContext(ctx, requestMethod, postInfo.URL, payload)
 		if err != nil {
-			network.logger.Error(fmt.Sprintf(`400 Unable to construct "%s" request for URL : "%s"`, requestMethod, postInfo.URL))
+			network.logger.Error(fmt.Sprintf(`400 Unable to construct %q request for URL : %q`, requestMethod, postInfo.URL))
```
Comment (lvrach) [inline router/network.go:137]: [super-minor] backticks no longer needed
```suggestion
			network.logger.Error(fmt.Sprintf("400 Unable to construct %q request for URL : %q", requestMethod, postInfo.URL))
```

**Review by lvrach — APPROVED** [review]

**Pull request merged**